### PR TITLE
Implemented fcntl and refactoring configure.ac

### PR DESCRIPTION
**Pull request by catap**
Supported commands from:
 - UNIX93...Open Group Base Specifications Issue 7
 - BSD 2.10 (as example of very old UNIX)
 - macOS 10.12 man-page
 - linux man-page 4.11

Refactoring:
 - switch configure at ptrace features to use `AC_CHECK_DECLS`/`AC_CHECK_TYPES`

### CODE REVIEW

**Comment by riataman** [comment]
Would be nice to change the cases where its operating in a file descriptor to PARAMETER_TYPE_FILE_DESCRIPTOR so that the extra information on the descriptor can be printed).

File: file.c [hunk 1]
```diff
@@ -603,3 +603,436 @@ void RETRACE_IMPLEMENTATION(strmode)(int mode, char *bp)
...
+    retrace_log_and_redirect_before(&event_info);
+
+    switch (cmd) {
+	// hack for duplicating FD
+	case F_DUPFD:
+#if HAVE_DECL_F_DUPFD_CLOEXEC
+	case F_DUPFD_CLOEXEC:
+#endif
+	    r = real_fcntl(fildes, cmd, int_parameter);
+	    if (r > 0) {
```
Comment (drystone) [inline file.c]: `if (r >= 0) { ... }`
Unlikely, I know, but just possible `r == 0`

**Comment by catap** [comment]
@riataman each call of `fcntl` is operated with  a file descriptor I guess ;)

**Comment by catap** [comment]
@riataman, @drystone, fixed at https://github.com/riboseinc/retrace/pull/176/commits/a4e732e14087ddbb65acd8e56b5db066b92beb0f

**Comment by ronaldtse** [comment]
@catap the current build [fails](https://travis-ci.org/riboseinc/retrace/jobs/249135170) due to code formatting:

* ERROR: switch and case should be at the same indent, #410: FILE: file.c:689:
* ERROR: switch and case should be at the same indent, #592: FILE: file.c:871:

Could you fix it so we can merge it? Thanks!

**Comment by catap** [comment]
@ronaldtse 
It has only style warnings likes
```
WARNING: Possible switch case/default not preceded by break or fallthrough comment
#608: FILE: file.c:894:
+	case F_RDAHEAD:

WARNING: Possible switch case/default not preceded by break or fallthrough comment
#611: FILE: file.c:897:
+	case F_NOCACHE:

WARNING: Possible switch case/default not preceded by break or fallthrough comment
#614: FILE: file.c:900:
+	case F_SETNOSIGPIPE:

```

I have no idea how to solve it. Can we merge?

**Comment by riataman** [comment]
I started writing the fall trough comments that the script wanted, but after doing three I say to myself, this is not worth it ...

**Pull request merged**